file path,line #,comment,satd
compose/setup.py,1,!/usr/bin/env python,not
compose/setup.py,2,-*- coding: utf-8 -*-,not
compose/contrib/migration/migrate-compose-file-v1-to-v2.py,1,!/usr/bin/env python,not
compose/contrib/migration/migrate-compose-file-v1-to-v2.py,77,"""container:<service name>"" is now ""service:<service name>""",not
compose/script/test/versions.py,1,!/usr/bin/env python,not
compose/script/test/versions.py,67,non-GA releases should appear before GA releases,SATD
compose/script/test/versions.py,68,Order: tp -> beta -> rc -> GA,not
compose/script/test/versions.py,85,List of versions known to be broken and should not be used,not
compose/script/release/release.py,1,!/usr/bin/env python3,not
compose/tests/conftest.py,8,FIXME Skipping all the acceptance tests when in `--conformity`,SATD
compose/tests/__init__.py,7,NOQA,not
compose/tests/__init__.py,9,NOQA,not
compose/tests/__init__.py,14,NOQA,not
compose/tests/acceptance/context_test.py,1,-*- coding: utf-8 -*-,not
compose/tests/acceptance/cli_test.py,1,-*- coding: utf-8 -*-,not
compose/tests/acceptance/cli_test.py,175,Hack: allow project to be overridden,SATD
compose/tests/acceptance/cli_test.py,184,Remove once Hijack and CloseNotifier sign a peace treaty,not
compose/tests/acceptance/cli_test.py,197,Prevent tearDown from trying to create a project,not
compose/tests/acceptance/cli_test.py,289,assert there are no python objects encoded in the output,not
compose/tests/acceptance/cli_test.py,350,{'name': 'networks_foo'},not
compose/tests/acceptance/cli_test.py,755,Make sure we have the latest busybox already,not
compose/tests/acceptance/cli_test.py,762,"If previous build happened on another node, cache won't be available",not
compose/tests/acceptance/cli_test.py,767,Make sure we have the latest busybox already,not
compose/tests/acceptance/cli_test.py,848,96 * 1024 * 1024,not
compose/tests/acceptance/cli_test.py,1078,Ensure containers don't have stdin and stdout connected in -d mode,not
compose/tests/acceptance/cli_test.py,1091,Ensure containers don't have stdin and stdout connected in -d mode,not
compose/tests/acceptance/cli_test.py,1166,The code below is a Swarm-compatible equivalent to volume.exists(),not
compose/tests/acceptance/cli_test.py,1224,Two networks were created: back and front,not
compose/tests/acceptance/cli_test.py,1252,One network was created: internal,not
compose/tests/acceptance/cli_test.py,1271,One networks was created: front,not
compose/tests/acceptance/cli_test.py,1294,Two networks were created: back and front,not
compose/tests/acceptance/cli_test.py,1297,lookup by ID instead of name in case of duplicates,SATD
compose/tests/acceptance/cli_test.py,1313,db and app joined the back network,not
compose/tests/acceptance/cli_test.py,1316,web and app joined the front network,not
compose/tests/acceptance/cli_test.py,1319,web can see app but not db,not
compose/tests/acceptance/cli_test.py,1323,app can see db,not
compose/tests/acceptance/cli_test.py,1326,"app has aliased db to ""database""",not
compose/tests/acceptance/cli_test.py,1480,No network was created,not
compose/tests/acceptance/cli_test.py,1489,console was not started,not
compose/tests/acceptance/cli_test.py,1494,web has links,not
compose/tests/acceptance/cli_test.py,1771,env overridden,not
compose/tests/acceptance/cli_test.py,1773,keep environment from yaml,not
compose/tests/acceptance/cli_test.py,1775,added option from command line,not
compose/tests/acceptance/cli_test.py,1785,Ensure stdin/out was open,not
compose/tests/acceptance/cli_test.py,1978,env overridden,not
compose/tests/acceptance/cli_test.py,1980,keep environment from yaml,not
compose/tests/acceptance/cli_test.py,1982,added option from command line,not
compose/tests/acceptance/cli_test.py,1984,make sure a value with a = don't crash out,not
compose/tests/acceptance/cli_test.py,1988,create one off container,not
compose/tests/acceptance/cli_test.py,1993,get port information,not
compose/tests/acceptance/cli_test.py,1997,close all one off containers we just created,not
compose/tests/acceptance/cli_test.py,2000,check the ports,not
compose/tests/acceptance/cli_test.py,2005,create one off container,not
compose/tests/acceptance/cli_test.py,2010,get port information,not
compose/tests/acceptance/cli_test.py,2015,close all one off containers we just created,not
compose/tests/acceptance/cli_test.py,2018,check the ports,not
compose/tests/acceptance/cli_test.py,2025,create one off container,not
compose/tests/acceptance/cli_test.py,2030,get port information,not
compose/tests/acceptance/cli_test.py,2034,close all one off containers we just created,not
compose/tests/acceptance/cli_test.py,2037,check the ports,not
compose/tests/acceptance/cli_test.py,2042,create one off container,not
compose/tests/acceptance/cli_test.py,2052,get port information,not
compose/tests/acceptance/cli_test.py,2056,close all one off containers we just created,not
compose/tests/acceptance/cli_test.py,2059,check the ports,not
compose/tests/acceptance/cli_test.py,2064,create one off container,not
compose/tests/acceptance/cli_test.py,2071,exposed ports are not mapped to host ports,not
compose/tests/acceptance/cli_test.py,2082,close all one off containers we just created,not
compose/tests/acceptance/cli_test.py,2155,"TODO: once we drop support for API <1.24, this can be changed to:",SATD
compose/tests/acceptance/cli_test.py,2156,assert config['Aliases'] == [container.short_id],not
compose/tests/acceptance/cli_test.py,2175,"TODO: once we drop support for API <1.24, this can be changed to:",SATD
compose/tests/acceptance/cli_test.py,2176,assert config['Aliases'] == [container.short_id],not
compose/tests/acceptance/cli_test.py,2225,os.environ doesn't support unicode values in Py2,not
compose/tests/acceptance/cli_test.py,2227,... and doesn't support byte values in Py3,not
compose/tests/acceptance/cli_test.py,2232,Can't retrieve output on Py3. See issue #3670,not
compose/tests/acceptance/cli_test.py,2405,Sometimes logs are picked up from old containers that haven't yet,not
compose/tests/acceptance/cli_test.py,2406,been removed (removal in Swarm is async),not
compose/tests/acceptance/cli_test.py,2517,The container is still running. It has only been paused,not
compose/tests/acceptance/cli_test.py,2789,"Sort by name so we get [db, web]",not
compose/tests/acceptance/cli_test.py,2840,SIGKILL,not
compose/tests/integration/state_test.py,415,update the image,not
compose/tests/integration/state_test.py,418,"In the case of a cluster, there's a chance we pick up the old image when",not
compose/tests/integration/state_test.py,419,"calculating the new hash. To circumvent that, untag the old image first",not
compose/tests/integration/state_test.py,420,See also: https://github.com/moby/moby/issues/26852,not
compose/tests/integration/service_test.py,297,"Match the last component (""host-path""), because boot2docker symlinks /tmp",not
compose/tests/integration/service_test.py,371,Ensure mounts are converted to volumes if API version < 1.30,not
compose/tests/integration/service_test.py,372,Needed to support long syntax in the 3.2 format,not
compose/tests/integration/service_test.py,388,Ensure tmpfs mounts are converted to tmpfs entries if API version < 1.30,not
compose/tests/integration/service_test.py,389,Needed to support long syntax in the 3.2 format,not
compose/tests/integration/service_test.py,431,https://github.com/docker/compose/issues/6280,not
compose/tests/integration/service_test.py,513,reload volume data,not
compose/tests/integration/service_test.py,533,"In Swarm, the env marker is consumed and the container should be deployed",not
compose/tests/integration/service_test.py,534,on the same node.,not
compose/tests/integration/service_test.py,574,reload volume data,not
compose/tests/integration/service_test.py,577,Do this twice to reproduce the bug,SATD
compose/tests/integration/service_test.py,587,"In Swarm, the env marker is consumed and the container should be deployed",not
compose/tests/integration/service_test.py,588,on the same node.,not
compose/tests/integration/service_test.py,605,reload volume data,not
compose/tests/integration/service_test.py,608,Do this twice to reproduce the bug,SATD
compose/tests/integration/service_test.py,619,"In Swarm, the env marker is consumed and the container should be deployed",not
compose/tests/integration/service_test.py,620,on the same node.,not
compose/tests/integration/service_test.py,834,retrieve new name from server,not
compose/tests/integration/service_test.py,1139,FIXME: remove once support is added,SATD
compose/tests/integration/service_test.py,1276,"self.assertEqual(container['NetworkSettings']['Ports'], {",not
compose/tests/integration/service_test.py,1277,'8000/tcp': [,not
compose/tests/integration/service_test.py,1278,{,not
compose/tests/integration/service_test.py,1279,"'HostIp': '127.0.0.1',",not
compose/tests/integration/service_test.py,1280,"'HostPort': '8001',",not
compose/tests/integration/service_test.py,1281,"},",not
compose/tests/integration/service_test.py,1282,"],",not
compose/tests/integration/service_test.py,1283,'9000/udp': [,not
compose/tests/integration/service_test.py,1284,{,not
compose/tests/integration/service_test.py,1285,"'HostIp': '0.0.0.0',",not
compose/tests/integration/service_test.py,1286,"'HostPort': '9001',",not
compose/tests/integration/service_test.py,1287,"},",not
compose/tests/integration/service_test.py,1288,"],",not
compose/tests/integration/service_test.py,1289,}),not
compose/tests/integration/service_test.py,1302,Ensure containers don't have stdout or stdin connected,not
compose/tests/integration/service_test.py,1640,TODO: remove this test once minimum docker version is 1.8.x,SATD
compose/tests/integration/testcases.py,181,"If the Engine is not part of a Swarm, the SDK will raise",not
compose/tests/integration/testcases.py,182,an APIError,not
compose/tests/integration/testcases.py,186,Only make the API call if the value hasn't been cached yet,not
compose/tests/integration/environment_test.py,51,Note that the warning silencing and the env variables check is,not
compose/tests/integration/environment_test.py,52,done in `project_from_options`,not
compose/tests/integration/environment_test.py,53,"So no need to have a proper options map, the `COMMAND` key is enough",not
compose/tests/integration/project_test.py,379,"self.assertEqual(len(project.containers()), 0)",not
compose/tests/integration/project_test.py,620,https://github.com/docker/compose/issues/4751,not
compose/tests/integration/project_test.py,656,"does scale=0 ,makes any sense? after recreating at least 1 container is running",not
compose/tests/integration/project_test.py,1519,Add the secret,not
compose/tests/integration/project_test.py,1527,First up with non-secret,not
compose/tests/integration/project_test.py,1535,Then up with secret,not
compose/tests/integration/project_test.py,1697,Use composetest_ prefix so it gets garbage-collected in tearDown(),not
compose/tests/unit/cli_test.py,1,encoding: utf-8,not
compose/tests/unit/cli_test.py,74,Environment has priority over .env file,not
compose/tests/unit/cli_test.py,174,"NOTE: The ""run"" command is supposed to be a one-off tool; therefore restart policy ""no""",not
compose/tests/unit/cli_test.py,175,"(the default) is enforced despite explicit wish for ""always"" in the project",not
compose/tests/unit/cli_test.py,176,configuration file,not
compose/tests/unit/parallel_test.py,56,we should always get the lock because we're the only thread running,not
compose/tests/unit/parallel_test.py,80,we should always get the lock because we're the only thread running,not
compose/tests/unit/progress_stream_test.py,1,~*~ encoding: utf-8 ~*~,not
compose/tests/unit/utils_test.py,1,encoding: utf-8,not
compose/tests/unit/service_test.py,1033,This is aimed at simulating a race between the API call to list the,not
compose/tests/unit/service_test.py,1034,"containers, and the ones to inspect each of the listed containers.",not
compose/tests/unit/service_test.py,1035,It can happen that a container has been removed after we listed it.,not
compose/tests/unit/service_test.py,1037,containers() returns a container that is about to be removed,not
compose/tests/unit/service_test.py,1042,inspect_container() will raise a NotFound when trying to inspect,not
compose/tests/unit/service_test.py,1043,"rm_cont_id, which at this point has been removed",not
compose/tests/unit/service_test.py,1059,We should nonetheless be able to create a new container,not
compose/tests/unit/service_test.py,1065,Trigger 3.2 format workaround,SATD
compose/tests/unit/project_test.py,1,encoding: utf-8,not
compose/tests/unit/project_test.py,332,Assert the return value is a generator,not
compose/tests/unit/project_test.py,489,Assert the return value is a generator,not
compose/tests/unit/project_test.py,792,"`stop_grace_period` is not set to 'web' service,",not
compose/tests/unit/project_test.py,793,then it is stopped with the default timeout.,not
compose/tests/unit/project_test.py,797,"`stop_grace_period` is set to 'db' service,",not
compose/tests/unit/project_test.py,798,then it is stopped with the specified timeout and,not
compose/tests/unit/project_test.py,799,the value is not overridden by the previous function call.,not
compose/tests/unit/cli/command_test.py,1,~*~ encoding: utf-8 ~*~,not
compose/tests/unit/cli/docker_client_test.py,214,verify is a special case - since `--tlsverify` = False means it,not
compose/tests/unit/cli/docker_client_test.py,215,"wasn't used, we set it if either the environment or the flag is True",not
compose/tests/unit/cli/docker_client_test.py,216,see https://github.com/docker/compose/issues/5632,not
compose/tests/unit/config/interpolation_test.py,1,encoding: utf-8,not
compose/tests/unit/config/interpolation_test.py,457,Test regression on docker/compose#5829,not
compose/tests/unit/config/config_test.py,1,encoding: utf-8,not
compose/tests/unit/config/config_test.py,902,service.dockerfile is invalid in v2,not
compose/tests/unit/config/config_test.py,1054,If build argument is None then it will be converted to the empty,not
compose/tests/unit/config/config_test.py,1055,"string. Make sure that int zero kept as it is, i.e. not converted to",not
compose/tests/unit/config/config_test.py,1056,the empty string,not
compose/tests/unit/config/config_test.py,2398,Regression: https://github.com/docker/compose/issues/4184,not
compose/tests/unit/config/config_test.py,3383,Valid port mappings ARE NOT valid 'expose' entries,not
compose/tests/unit/config/config_test.py,3391,Valid single ports ARE valid 'expose' entries,not
compose/tests/unit/config/config_test.py,5432,Values coming from env_files are not allowed to have variables,not
compose/tests/unit/config/environment_test.py,1,encoding: utf-8,not
compose/compose/container.py,93,One-off containers are no longer assigned numbers and use slugs instead.,not
compose/compose/container.py,330,inspect,not
compose/compose/container.py,333,ps,not
compose/compose/parallel.py,59,This is to bubble up ImageNotFound exceptions to the client so we,not
compose/compose/parallel.py,60,can prompt the user if they want to rebuild.,not
compose/compose/parallel.py,185,See https://github.com/docker/compose/issues/189,not
compose/compose/parallel.py,295,move up,SATD
compose/compose/parallel.py,297,erase,not
compose/compose/parallel.py,301,move back down,SATD
compose/compose/volume.py,199,These options are set internally,not
compose/compose/volume.py,209,We are only interested in user-specified labels,not
compose/compose/project.py,299,actually returning inversed dependencies,not
compose/compose/project.py,424,Only for v1 files or when Compose is forced to use an older API version,not
compose/compose/project.py,448,This is a guard against some events broadcasted by swarm that,not
compose/compose/project.py,449,don't have a status field.,not
compose/compose/project.py,450,See https://github.com/docker/compose/issues/3316,not
compose/compose/project.py,455,this can fail if the container has been removed or if the event,not
compose/compose/project.py,456,refers to an image,not
compose/compose/project.py,466,"New, better event API was introduced in 1.22.",SATD
compose/compose/project.py,480,Container may have been removed (e.g. if this is a destroy event),not
compose/compose/project.py,501,TODO: support other event types,SATD
compose/compose/project.py,655,Attempting to pull service with no `image` key is a no-op,not
compose/compose/project.py,694,Considering <image> and <image:latest> as the same,not
compose/compose/project.py,877,"Compose file already uses a digest, no lookup required",not
compose/compose/project.py,896,TODO: pick a digest based on the image tag if there are multiple,SATD
compose/compose/project.py,897,digests,not
compose/compose/project.py,902,Fetch the image digest from the registry,not
compose/compose/service.py,283,do nothing as we already have the desired number,not
compose/compose/service.py,291,"we have some stopped containers, check for divergences",not
compose/compose/service.py,296,Remove containers that have diverged,not
compose/compose/service.py,329,This is only necessary for `scale` and `volumes_from`,not
compose/compose/service.py,330,auto-creating containers to satisfy the dependency.,not
compose/compose/service.py,550,"The create action needs always needs an initial scale, but otherwise,",not
compose/compose/service.py,551,we set scale to none in no-rescale scenarios (`run` dependencies),not
compose/compose/service.py,557,Updating the image ID on the container object lets us recover old volumes if,not
compose/compose/service.py,558,the new image uses them as well,not
compose/compose/service.py,737,"The config schema already prevents this, but it might be",not
compose/compose/service.py,738,bypassed if Compose is called programmatically.,not
compose/compose/service.py,772,Account for containers that might have been removed since we fetched,not
compose/compose/service.py,773,the list.,not
compose/compose/service.py,859,"If a qualified hostname was given, split it into an",not
compose/compose/service.py,860,unqualified hostname and a domainname unless domainname,not
compose/compose/service.py,861,was also given explicitly. This matches behavior,not
compose/compose/service.py,862,until Docker Engine 1.11.0 - Docker API 1.23.,not
compose/compose/service.py,911,Delete options which are only used in HostConfig,not
compose/compose/service.py,946,Workaround for 3.2 format,SATD
compose/compose/service.py,966,Remove possible duplicates (see e.g. https://github.com/docker/compose/issues/5885).,not
compose/compose/service.py,967,unique_everseen preserves order. (see https://github.com/docker/compose/issues/6091).,not
compose/compose/service.py,1114,Ensure the HTTP connection is not reused for another,not
compose/compose/service.py,1115,"streaming command, as the Docker daemon can sometimes",not
compose/compose/service.py,1116,complain about it,not
compose/compose/service.py,1193,there are no external bindings,not
compose/compose/service.py,1197,we only need to check the first binding from the range,not
compose/compose/service.py,1200,non-tuple binding means there is a host port specified,not
compose/compose/service.py,1204,"extract actual host port from tuple of (host_ip, host_port)",not
compose/compose/service.py,1415,Names,not
compose/compose/service.py,1427,Images,not
compose/compose/service.py,1456,Volumes,not
compose/compose/service.py,1509,No need to preserve host volumes,not
compose/compose/service.py,1513,Attempting to rebind tmpfs volumes breaks: https://github.com/docker/compose/issues/4751,not
compose/compose/service.py,1519,"New volume, doesn't exist in the old container",not
compose/compose/service.py,1523,"Volume was previously a host volume, now it's a container volume",not
compose/compose/service.py,1527,Volume (probably an image volume) is overridden by a mount in the service's config,not
compose/compose/service.py,1528,and would cause a duplicate mountpoint error,not
compose/compose/service.py,1532,Copy existing volume from old container,not
compose/compose/service.py,1608,Labels,not
compose/compose/service.py,1627,Ulimits,not
compose/compose/service.py,1654,TODO: remove once fix is available in docker-py,SATD
compose/compose/service.py,1666,Ports,not
compose/compose/service.py,1710,"python2 os.stat() doesn't support unicode on some UNIX, so we",not
compose/compose/service.py,1711,encode it to a bytestring to be safe,not
compose/compose/service.py,1812,Fix non ascii chars on Python2. To remove when #6890 is complete.,not
compose/compose/service.py,1824,In case of `DOCKER_BUILDKIT=1`,not
compose/compose/service.py,1825,there is no success message already present in the output.,not
compose/compose/service.py,1826,Since that's the way `Service::build` gets the `image_id`,not
compose/compose/service.py,1827,it has to be added `manually`,not
compose/compose/timeparse.py,1,!/usr/bin/env python,not
compose/compose/timeparse.py,2,-*- coding: utf-8 -*-,not
compose/compose/timeparse.py,13,MIT LICENSE,not
compose/compose/timeparse.py,14,,not
compose/compose/timeparse.py,15,"Permission is hereby granted, free of charge, to any person",not
compose/compose/timeparse.py,16,obtaining a copy of this software and associated documentation files,not
compose/compose/timeparse.py,17,"(the ""Software""), to deal in the Software without restriction,",not
compose/compose/timeparse.py,18,"including without limitation the rights to use, copy, modify, merge,",not
compose/compose/timeparse.py,19,"publish, distribute, sublicense, and/or sell copies of the Software,",not
compose/compose/timeparse.py,20,"and to permit persons to whom the Software is furnished to do so,",not
compose/compose/timeparse.py,21,subject to the following conditions:,not
compose/compose/timeparse.py,22,,not
compose/compose/timeparse.py,23,The above copyright notice and this permission notice shall be,not
compose/compose/timeparse.py,24,included in all copies or substantial portions of the Software.,not
compose/compose/timeparse.py,25,,not
compose/compose/timeparse.py,26,"THE SOFTWARE IS PROVIDED ""AS IS"", WITHOUT WARRANTY OF ANY KIND,",not
compose/compose/timeparse.py,27,"EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF",not
compose/compose/timeparse.py,28,"MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND",not
compose/compose/timeparse.py,29,NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS,not
compose/compose/timeparse.py,30,"BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN",not
compose/compose/timeparse.py,31,"ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN",not
compose/compose/timeparse.py,32,CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE,not
compose/compose/timeparse.py,33,SOFTWARE.,not
compose/compose/progress_stream.py,37,"if it's a progress event and we have a terminal, then display the progress bars",not
compose/compose/progress_stream.py,48,move cursor up `diff` rows,SATD
compose/compose/progress_stream.py,54,move cursor back down,SATD
compose/compose/progress_stream.py,67,erase current line,not
compose/compose/network.py,50,Swarm nodes do not register overlay networks that were,not
compose/compose/network.py,51,created on a different node unless they're in use.,not
compose/compose/network.py,52,See docker/compose#4399,not
compose/compose/network.py,231,We are only interested in user-specified labels,not
compose/compose/network.py,332,Ensure Compose will pick a consistent primary network if no,not
compose/compose/network.py,333,priority is set,not
compose/compose/const.py,46,minimum DOCKER ENGINE API version needed to support,not
compose/compose/const.py,47,features for each compose schema version,not
compose/compose/cli/signals.py,35,on Windows a ValueError will be raised if trying to set signal handler for SIGHUP,not
compose/compose/cli/signals.py,41,Restore default behavior for SIGPIPE instead of raising,not
compose/compose/cli/signals.py,42,an exception when encountered.,not
compose/compose/cli/formatter.py,20,get_terminal_size can't determine the size if compose is piped,not
compose/compose/cli/formatter.py,21,to another command. But in such case it doesn't make sense to,not
compose/compose/cli/formatter.py,22,try format the output by terminal size as this output is consumed,not
compose/compose/cli/formatter.py,23,by another command. So let's pretend we have a huge terminal so,not
compose/compose/cli/formatter.py,24,output is single-lined,not
compose/compose/cli/main.py,116,Skip looking up the compose file.,not
compose/compose/cli/main.py,136,Disable requests logging,not
compose/compose/cli/main.py,176,stolen from docopt master,not
compose/compose/cli/main.py,1285,Ensure that run command remains one-off (issue #6302),not
compose/compose/cli/main.py,1573,UCP does multi-node scheduling with traditional Compose files.,not
compose/compose/cli/utils.py,19,WindowsError is not defined on non-win32 platforms. Avoid runtime errors by,not
compose/compose/cli/utils.py,20,defining it as OSError (its parent class) if missing.,not
compose/compose/cli/utils.py,67,"On Windows, subprocess.call() can still raise exceptions. Normalize",not
compose/compose/cli/utils.py,68,to POSIXy behaviour by returning a nonzero exit code.,not
compose/compose/cli/errors.py,72,https://github.com/docker/compose/issues/5005,not
compose/compose/cli/errors.py,104,They've set a custom API version,not
compose/compose/cli/errors.py,129,https://github.com/docker/compose/issues/5442,not
compose/compose/cli/errors.py,130,Ignore the error and print the generic message instead.,not
compose/compose/cli/log_printer.py,93,Returning the name of the container that started the,not
compose/compose/cli/log_printer.py,94,the cascade_stop so we can return the correct exit code,not
compose/compose/cli/log_printer.py,99,"There are no running containers left to tail, so exit",not
compose/compose/cli/log_printer.py,101,"We got an empty line because of a timeout, but there are still",not
compose/compose/cli/log_printer.py,102,"active containers to tail, so continue",not
compose/compose/cli/log_printer.py,111,This may happen if the user's locale settings don't support UTF-8,not
compose/compose/cli/log_printer.py,112,and UTF-8 characters are present in the log line. The following,not
compose/compose/cli/log_printer.py,113,"will output a ""degraded"" log with unsupported characters",not
compose/compose/cli/log_printer.py,114,replaced by `?`,not
compose/compose/cli/log_printer.py,137,"Container order is unspecified, so they are sorted by name in order to make",not
compose/compose/cli/log_printer.py,138,container:presenter (log color) assignment deterministic when given a list of containers,not
compose/compose/cli/log_printer.py,139,with the same names.,not
compose/compose/cli/log_printer.py,188,if the container doesn't have a log_stream we need to attach to container,not
compose/compose/cli/log_printer.py,189,before log printer starts running,not
compose/compose/cli/log_printer.py,231,"Container was stopped and started, we need a new thread",not
compose/compose/cli/log_printer.py,234,Container crashed so we should reattach to it,not
compose/compose/cli/log_printer.py,241,Just ignore errors when reattaching to already crashed containers,not
compose/compose/cli/log_printer.py,260,See https://github.com/docker/compose/issues/189,not
compose/compose/cli/docopt_command.py,43,"we certainly want to have ""exec"" command, since that's what docker client has",not
compose/compose/cli/docopt_command.py,44,but in python exec is a keyword,not
compose/compose/cli/command.py,50,get the context for the run,not
compose/compose/cli/docker_client.py,84,verify is a special case - with docopt `--tlsverify` = False means it,not
compose/compose/cli/docker_client.py,85,"wasn't used, so we set it if either the environment or the flag is True",not
compose/compose/cli/docker_client.py,86,see https://github.com/docker/compose/issues/5632,not
compose/compose/cli/docker_client.py,91,FIXME: Modify TLSConfig to take a cert_path argument and do this internally,SATD
compose/compose/cli/docker_client.py,97,Default location for cert files is ~/.docker,not
compose/compose/cli/docker_client.py,108,--tls is a noop,not
compose/compose/cli/docker_client.py,136,check env for DOCKER_HOST and certs path,not
compose/compose/cli/docker_client.py,162,Workaround for,SATD
compose/compose/cli/docker_client.py,163,https://pyinstaller.readthedocs.io/en/v3.3.1/runtime-information.html#ld-library-path-libpath-considerations,not
compose/compose/config/serialize.py,34,"Empirically only y/n appears to be an issue, but this might change",not
compose/compose/config/serialize.py,35,depending on which PyYaml version is being used. Err on safe side.,not
compose/compose/config/serialize.py,86,"For compatibility mode, this option is invalid in v2",not
compose/compose/config/sort_services.py,47,Topological sort (Cormen/Tarjan algorithm).,not
compose/compose/config/config.py,1391,Shortcut where specifying a string is treated as the build context,not
compose/compose/config/config.py,1423,"We have a build section but no context, so nothing to validate",not
compose/compose/config/config.py,1529,Sometimes the user's locale sets an encoding that doesn't match,not
compose/compose/config/config.py,1530,"the YAML files. Im such cases, retry once with the ""default""",not
compose/compose/config/config.py,1531,UTF-8 encoding,not
compose/compose/config/types.py,27,TODO: drop service_names arg when v1 is removed,SATD
compose/compose/config/types.py,121,TODO: validate string contains ':' ?,SATD
compose/compose/config/types.py,144,"If a path looks like a UNIX absolute path on Windows, it probably is;",not
compose/compose/config/types.py,145,we'll need to revert the backslashes to forward slashes after normalization,not
compose/compose/config/types.py,245,"relative paths in windows expand to include the drive, eg C:\",not
compose/compose/config/types.py,246,so we join the first 2 parts back together to count as one,not
compose/compose/config/types.py,367,"""x-y:z"" format",not
compose/compose/config/types.py,387,"When extending a service with ports, the port definitions have already been parsed",not
compose/compose/config/types.py,482,based on https://github.com/docker/cli/blob/9de1b162f/cli/command/container/opts.go#L673-L697,not
compose/compose/config/environment.py,78,"Values from the command line take priority, unless they're unset",not
compose/compose/config/environment.py,79,in which case they take the value from the system's environment,not
compose/compose/config/environment.py,121,"Convert a value to a boolean using ""common sense"" rules.",not
compose/compose/config/environment.py,122,"Unset, empty, ""0"" and ""false"" (i-case) yield False.",not
compose/compose/config/environment.py,123,All other values yield True.,not
compose/compose/config/__init__.py,1,flake8: noqa,not
compose/compose/config/validation.py,277,The service_name is one of the keys in the json object,not
compose/compose/config/validation.py,497,TODO: why is this necessary?,SATD
compose/compose/config/validation.py,530,"when disable is true config.py::process_healthcheck adds ""test: ['NONE']"" to service_config",not
compose/compose/config/interpolation.py,127,Modified from python2.7/string.py,not
compose/compose/config/interpolation.py,129,Helper function for .sub(),not
compose/compose/config/interpolation.py,194,We must be able to handle octal representation for `mode` values notably,not
